application/controllers/admin/SurveymenuController.php

```diff
@@ -284,52 +285,27 @@ public function delete()
      */
     public function restore()
     {
+        $this->requirePostRequest();
+
         if (!(Permission::model()->hasGlobalPermission('settings', 'delete') && Permission::model()->hasGlobalPermission('settings', 'update'))) {
             Yii::app()->user->setFlash('error', gT("Access denied"));
             $this->getController()->redirect(Yii::app()->createUrl('/admin'));
         }
 
         $model = Surveymenu::model();
-        if (Yii::app()->request->isPostRequest) {
```

Comment:
This part didn't force POST ? We have surely some other place where it's like this, no ?